File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/events/EventQueryService.java
package net.nemerosa.ontrack.model.events;

import net.nemerosa.ontrack.model.structure.ID;
import net.nemerosa.ontrack.model.structure.ProjectEntityType;
import net.nemerosa.ontrack.model.structure.Signature;

import java.util.List;
import java.util.Optional;

/**
 * Service used to get access to the list of events.
 */
public interface EventQueryService {

    List<Event> getEvents(int offset, int count);

    List<Event> getEvents(ProjectEntityType entityType, ID entityId, int offset, int count);

    List<Event> getEvents(ProjectEntityType entityType, ID entityId, EventType eventType, int offset, int count);

    Optional<Signature> getLastEventSignature(ProjectEntityType entityType, ID entityId, EventType eventType);

}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/events/EventQueryService.kt
package net.nemerosa.ontrack.model.events

import net.nemerosa.ontrack.model.structure.ID
import net.nemerosa.ontrack.model.structure.ProjectEntity
import net.nemerosa.ontrack.model.structure.ProjectEntityType
import net.nemerosa.ontrack.model.structure.Signature
import java.util.*

/**
 * Service used to get access to the list of events.
 */
interface EventQueryService {

    fun getEvents(offset: Int, count: Int): List<Event>

    fun getEvents(entityType: ProjectEntityType, entityId: ID, offset: Int, count: Int): List<Event>

    fun getEvents(entityType: ProjectEntityType, entityId: ID, eventType: EventType, offset: Int, count: Int): List<Event>

    fun getLastEventSignature(entityType: ProjectEntityType, entityId: ID, eventType: EventType): Optional<Signature>

    fun getLastEvent(entityType: ProjectEntityType, entityId: ID, eventType: EventType): Optional<Event>

    fun getLastEvent(entity: ProjectEntity, eventType: EventType): Event?

}
